### PR TITLE
Add comment reflecting unexpected union cache behavior that affects validation order

### PR DESCRIPTION
**Pull request by sydney-runkle**

### CODE REVIEW

**Comment by sydney-runkle** [comment]
For some reason this seems to be failing now given the added benchmarks, which is quite odd.

**Comment by sydney-runkle** [comment]
Huh, this is actually caused by: https://github.com/python/cpython/issues/103749, which is a bit out of our control at the moment.

**Comment by sydney-runkle** [comment]
See:

Hmm, odd Python behavior:
```py
from typing import Union, List

class Model1:
    v: List[Union[str, int]]

class Model2:
    v: List[Union[int, str]]

print(Model1.__annotations__)
#> {'v': typing.List[typing.Union[str, int]]}
print(Model2.__annotations__)
# odd, this is normalized to be the same as above
#> {'v': typing.List[typing.Union[str, int]]}
```

But, without Model1

```py
from typing import Union, List

# class Model1:
#     v: List[Union[str, int]]

class Model2:
    v: List[Union[int, str]]

print(Model2.__annotations__)
# has the order you'd expect...
#> {'v': typing.List[typing.Union[int, str]]} 
```

File: tests/test_edge_cases.py [hunk 1]
```diff
@@ -497,7 +497,7 @@ class Model(BaseModel):
 
 def test_list_unions():
     class Model(BaseModel):
-        v: List[Union[int, str]] = ...
+        v: List[Union[int, int]] = ...
```
Comment (Viicos) [inline tests/test_edge_cases.py:500]: I would keep the union of int and str, and try to do an unordered list comparison (a la https://docs.python.org/3.5/library/unittest.html#unittest.TestCase.assertCountEqual but for pytest, don't know if it exists)

Comment (sydney-runkle) [reply tests/test_edge_cases.py:500]: Oops yes this annotation modification was a mistake.

**Review by Viicos — APPROVED** [review]

**Pull request merged**